cmake: Fixup shield overlay and conf file search locations

galak opened the pull request:
* Fix dir name when shield name doesn't match the dir name.  So for
  shield 'x_nucleo_iks01a3_shub' we will look in now for
  x_nucleo_iks01a3/boards/<BOARD>.{overlay,conf} instead of
  x_nucleo_iks01a3_shub/boards/<BOARD>.{overlay,conf}

* Add support to search for a board specific shield in
  <SHIELD DIR>/boards/<SHIELD>/<BOARD>.{overlay,conf}.  So we can
  support something like:

  x_nucleo_iks01a3/boards/x_nucleo_iks01a3_shub/stm32mp157c_dk2.overlay
  x_nucleo_iks01a3/boards/x_nucleo_iks01a3_shub/stm32mp157c_dk2.conf

Fixes #19078

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
cmake: add new search locations for shield overlay and conf files

* Add support to search for a board specific shield in
  <SHIELD DIR>/boards/<SHIELD>/<BOARD>.{overlay,conf}.  So we can
  support something like:

  x_nucleo_iks01a3/boards/x_nucleo_iks01a3_shub/stm32mp157c_dk2.overlay
  x_nucleo_iks01a3/boards/x_nucleo_iks01a3_shub/stm32mp157c_dk2.conf

Fixes #19078

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mike-scott commented on cmake/app/boilerplate.cmake:
```diff
@@ -307,38 +308,56 @@ foreach(root ${BOARD_ROOT})
+          ${shield_dir}/${s_dir}/dts_fixup.h
         )
+
+        # search for shield/boards/board.overlay file
```
I think if we're going to move the existing searches inside the `if (NOT _idx EQUAL -1)` block, then we do that in a single patch and explain the move.  Then a 2nd patch to use s_dir instead of s.

galak replied:
ok, split up the commits.

galak committed:
cmake: move shield board/conf handling

Move handling for shield boards/board.overlay, shield.conf and
boards/board.conf inside the SHIELD_LIST check.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
cmake: Fixup shield overlay and conf file search locations

* Fix dir name when shield name doesn't match the dir name.  So for
  shield 'x_nucleo_iks01a3_shub' we will look in now for
  x_nucleo_iks01a3/boards/<BOARD>.{overlay,conf} instead of
  x_nucleo_iks01a3_shub/boards/<BOARD>.{overlay,conf}

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mike-scott reviewed (approved):
LGTM.  I have not tested this change however.

erwango reviewed (approved):
I'm fine with the change. But we actually miss some doc on following topics:
- boards specific shields configurations
- shields revisions
- boards specific shields revisions configurations
I'll push a PR for this.
